federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -54,6 +58,13 @@ import (
 	"k8s.io/client-go/discovery"
 )
 
+const (
+	apiserverWaitTimeout   = 30 * 2 * time.Second
+	apiserverRetryInterval = 1 * time.Second
```

Comment:
This seems very short, and requires exponential backoff.  There are standard libraries to do the right thing here.